lib/aggregate/aggregate_store.rb

```diff
@@ -75,11 +85,24 @@ def changed?
       (defined?(super) && super) || @changed
     end
 
+    def saved_changes?
+      ActiveRecordHelpers::Version.if_version(
+        active_record_4: -> { raise NoMethodError, "undefined method 'saved_changes?' for #{self}" }
+      )
+      (defined?(super) && super) || @changed_on_save
+    end
+
     def set_changed
       @changed = aggregate_values != aggregate_initial_values
       aggregate_owner&.set_changed
     end
 
+    def set_saved_changes
```

Comment:
Does this match the Rails model expectations that `changes` gets reset after save?

Reply:
This function does not, but the reset of `changes` is called by the `after commit` callback in https://github.com/Invoca/aggregate/blob/62c63e463f78fa12fe44979fd6ca791a663982e9/lib/aggregate/container.rb#L23 
Is there a scenario where after save is called but after commit is not?

Reply:
Ah, then it might be fine. Forgot that was there..thanks!